controller/src/main/java/org/jboss/as/controller/ParallelBootOperationStepHandler.java

```diff
@@ -358,9 +366,10 @@ public void rollback() {}
                     }, new ModelNode());
                     return;
                 }
+
                 operationContext = new ParallelBootOperationContext(transactionControl, processState,
                         primaryContext, runtimeOps, controllingThread, controller, lockId, controller.getAuditLogger(),
-                        controller.getManagementModel().getRootResource(), extraValidationStepHandler);
+                        controller.getManagementModel().getRootResource(), extraValidationStepHandler, bootSecurityDomain::getAnonymousSecurityIdentity);
```

Comment:
This doesn't need to be fixed now but I think this should just be passing in a Supplier that provides the SecurityIdentity from primaryContext. That's the basic pattern for how the parallel boot context works - it relies on the primary context for everything possible.